_RFCs/0003-tracking-package-versions.md

```diff
@@ -0,0 +1,50 @@
+
+However, we do not have a process in place to handle edge releases of software. As of today, Docker 18 is the latest edge version of Docker, but the latest stable version is still Docker 17. This brought us to the question - should "core/docker" track the latest stable version or the latest edge version?
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Many Habitat packages track and package 3rd party software - this includes Docker, Node, Java, and more. We learned early on that even when a new major version of the software was released, many people still used the older version (and sometimes did not have the choice of upgrading immediately due to compliance or other reasons).
+
+Historically, when a new major version of software was released (i.e. when Node 8 became the new LTS over Node 7) we changed the "generic" package (i.e. "core/node") to track the newest major version and, for people who needed to use an older major version of the software, created a new plan (i.e. "core/node7") that they could refer to.
+
+When a new edge version of the software is released (i.e. Node 9.9.0 is the latest edge release as of this RFC), we do not track that in the generic package (i.e. core/node). The generic package must always track the latest stable release.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC would not change any current plans, it would apply to updated and new plans from the time of the adoption of the RFC.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Users who want the bleeding edge versions of software will not be able to get them through Habitat's core plans.
+
+The recommended way to use a non-stable version of a package is to create it as a part of your own origin, until it becomes stable, at which point it can be added to core.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+It would not make sense to add a new package for the edge version (i.e. if the latest edge release of Node is 10, create a "core/node10"), then remove that package when that version become the stable version and is tracked in core/node. It also would be confusing to, rather than remove the edge version package ("core/node10") when it becomes the LTS version, track Node 10 in both "core/node" and "core/node10".
+
+Some have mentioned the idea of not having any generic packages - rather than "core/node", have only "core/node8", "core/node9", etc. While possible, this would potentially break plans which already depend on "core/node" or "core/docker", and a large amount of existing plans, documentation, and more would be required to change. I don't feel it is worth that pain of making our users switch from something like "core/node" to always specifying the latest major versions, particulary for users who don't care about the version as long as it works.
```

Comment:
This point is particulary tricky indeed. The "as long as it work" have to be enforced somehow IMHO. 

Maybe when a new `X` major version gets out (which might not be retro-compatible then) we should create a new `core/packageX` and change the `core/package` to source `core/packageX`. This would allow people who want to be sure there won't be breakage to use latest stable anyways, instead of waiting for one's package to be broken when a new major release gets out.